datafusion-examples/examples/pruning.rs

```diff
@@ -187,7 +188,9 @@ impl PruningStatistics for MyCatalog {
 fn create_pruning_predicate(expr: Expr, schema: &SchemaRef) -> PruningPredicate {
     let df_schema = DFSchema::try_from(schema.as_ref().clone()).unwrap();
     let props = ExecutionProps::new();
-    let physical_expr = create_physical_expr(&expr, &df_schema, &props).unwrap();
+    let config_options = ConfigOptions::default();
```

Comment:
Something we could potentially do to make this API slightly easier to use might be to create a static default `ConfigOptions` 

Something like
```rust
impl ConfigOptions {
  /// returns a reference to default ConfigOptions
  pub fn default_singleton() -> &'static ConfigOptions 
}
```

This would then make it easier to return `&ConfigOptions` in various places when only the default was needed

For example, then in `LocalCsvTableFunc` you could avoid having to thread the `ConfigOptions` through as in that example having the actual config options isn't important